consensus/core/src/dag_state.rs

```diff
@@ -803,15 +850,17 @@ impl DagState {
 
         // Clean up old cached data. After flushing, all cached blocks are guaranteed to be persisted.
         let mut total_recent_refs = 0;
-        for (authority_refs, last_committed_round) in self
-            .recent_refs
-            .iter_mut()
-            .zip(self.last_committed_rounds.iter())
-        {
+
+        // First we store the evicted rounds for each authority.
```

Comment:
I think evicting first, then updating the evicted_round is more intuitive based on the name.

Reply:
I did it mostly to avoid the tricky situation with the ownership of `self` of having to hold mut and imut references, but refactored it to by pass this. Now will update the evicted rounds after doing the actual eviction.